solr/core/src/java/org/apache/solr/search/grouping/distributed/responseprocessor/TopGroupsShardResponseProcessor.java

```diff
@@ -188,7 +188,11 @@ public void process(ResponseBuilder rb, ShardRequest shardRequest) {
       }
       rb.mergedQueryCommandResults.put(query, new QueryCommandResult(mergedTopDocs, mergedMatches, maxScore));
     }
+    fillResultIds(rb);
+  }
+
 
+  static void fillResultIds(ResponseBuilder rb){
```

Comment:
https://issues.apache.org/jira/browse/SOLR-13576 opened for this fairly independent refactoring here.